tools/bin/commands/util/operator_funcs

```diff
@@ -20,25 +37,52 @@ dep_ensure() {
         dep ensure -vendor-only -v
     else
         echo "Running 'dep ensure' with $GO_BUILDER_IMAGE"
-        [ -d "dep-cache" ] || mkdir dep-cache
+        required_dirs
         local extra_opts=""
         if [ -n "$as_user" ]; then
             extra_opts="$extra_opts -u $as_user"
         fi
         docker run -w /gopath/src/github.com/syndesisio/syndesis/install/operator \
+                   -u $(id -u):$(id -g) \
                    $extra_opts \
+                   -v "$DOCKER_GOPATH_VOLUME":/gopath:z \
+                   -v "$DOCKER_CACHE_VOLUME":/.cache:z \
                    -v $operator_dir/dep-cache:/gopath/pkg/dep:z \
                    -v $operator_dir:/gopath/src/github.com/syndesisio/syndesis/install/operator:z \
                    $GO_BUILDER_IMAGE \
                    dep ensure -vendor-only -v
     fi
 }
 
+required_dirs() {
+    require_dir "$operator_dir/dep-cache"
+    require_dir "$DOCKER_CACHE_VOLUME"
+    require_dir "$DOCKER_GOPATH_VOLUME/pkg/dep"
+    require_dir "$DOCKER_GOPATH_VOLUME/src/github.com/syndesisio/syndesis/install/operator"
+
+    CLEANUP_DIRS+=($DOCKER_GOPATH_VOLUME)
+    CLEANUP_DIRS+=($DOCKER_CACHE_VOLUME)
+}
+
+require_dir() {
+    local dir=${1}
+    [ -d "$dir" ] || mkdir -p $dir
+    if [ "$(stat --format '%u:%g' $dir)" != "$(id -u)":"$(id -g)" ]; then
+        echo "ERROR: $dir is not owned by the current user"
+        exit 1
+    fi
+}
+
 gopath_dir() {
+    set +o nounset
     if [ -z "$GOPATH" ]; then
-        echo "ERROR: You have to set your GOPATH environment variable"
-        exit 1
+        if [ ! -d "$HOME/go" ]; then
```

Comment:
Never heard about that convention, but not harmful I guess ;-)

Reply:
This is how we hipsters roll, $GOPATH that so yesterday